pkg/controller/csr/csr_controller.go

```diff
@@ -0,0 +1,223 @@
+
+// Add creates a new CSR Controller and adds it to the Manager. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func Add(mgr manager.Manager, opts options.AddOptions) error {
+	if !opts.EnterpriseCRDExists {
+		// No need to start this controller (currently).
+		return nil
+	}
+	reconciler, err := newReconciler(mgr, opts)
+	if err != nil {
+		return err
+	}
+
+	ctrl, err := controller.New(controllerName, mgr, controller.Options{Reconciler: reconciler})
+	if err != nil {
+		return err
+	}
+
+	return add(mgr, ctrl)
+}
+
+// newReconciler returns a new *reconcile.Reconciler
+func newReconciler(mgr manager.Manager, opts options.AddOptions) (reconcile.Reconciler, error) {
+	clientset, err := kubernetes.NewForConfig(mgr.GetConfig())
+	if err != nil {
+		return nil, err
+	}
+
+	r := &ReconcileCSR{
+		client:             mgr.GetClient(),
+		scheme:             mgr.GetScheme(),
+		provider:           opts.DetectedProvider,
+		clusterDomain:      opts.ClusterDomain,
+		certificatesClient: clientset.CertificatesV1(),
+	}
+	return r, nil
+}
+
+// add adds watches for resources that are available at startup
+func add(_ manager.Manager, c controller.Controller) error {
+	return utils.AddCSRWatchWithLabel(c, LabelName)
+}
+
+// blank assignment to verify that ReconcileCompliance implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileCSR{}
+
+// ReconcileCSR reconciles CSRs objects
+type ReconcileCSR struct {
+	client             client.Client
+	scheme             *runtime.Scheme
+	provider           operatorv1.Provider
+	clusterDomain      string
+	certificatesClient typedcertificatesv1.CertificatesV1Interface
+}
+
+func (r ReconcileCSR) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling CSR Controller")
+	csrList := &certificatesv1.CertificateSigningRequestList{}
+
+	// Filter out unnecessary CSRs. (Calico CSRs are guaranteed to have this label).
+	requirement, err := labels.NewRequirement(LabelName, selection.Exists, []string{})
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	if err := r.client.List(ctx, csrList, &client.ListOptions{LabelSelector: labels.NewSelector().Add(*requirement)}); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	instance := &operatorv1.Installation{}
+	if err := r.client.Get(ctx, utils.DefaultInstanceKey, instance); err != nil {
+		if apierrors.IsNotFound(err) {
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, err
+	}
+
+	certificateManager, err := certificatemanager.Create(r.client, &instance.Spec, r.clusterDomain, common.OperatorNamespace(), certificatemanager.WithLogger(reqLogger))
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	for _, csr := range csrList.Items {
+		if csr.Spec.SignerName == utils.OperatorCSRSignerName && csr.Status.Certificate == nil {
```

Comment:
This would probably be better phrased as:

```go
if sigerName != operatorSigner || csr.Status.Certificate != nil {
  // Not for us, or already signed. 
  continue
}
```

To simplify indentation.